src/scripts/datadog-install.sh

```diff
@@ -12,33 +19,71 @@ log()
     echo \[$(date +%d%m%Y-%H:%M:%S)\] "$1" >> /var/log/datadog-install.log
 }
 
-if [ "$1" == "" ]; then
+#########################
+# Parameter handling
+#########################
+
+API_KEY=""
+NODE_ROLE=""
+
+#Loop through options passed
+while getopts :k:r:h optname; do
+  log "Option $optname set"
+  case $optname in
+    k) # DataDog API key
+      API_KEY="${OPTARG}"
+      ;;
+    r) # Node role
+      NODE_ROLE="${OPTARG}"
+      ;;
+    h) #show help
+      help
+      exit 2
+      ;;
+    \?) #unrecognized option - show help
+      echo -e \\n"ERROR: unknown option -${BOLD}$OPTARG${NORM}" >&2
+      help
+      exit 2
+      ;;
+  esac
+done
+
+if [ "$API_KEY" == "" ]; then
     log "Must provide a DataDog API Key" >&2
     exit 1
 fi
 
-if [ "$2" == "" ]; then
+if [ "$NODE_ROLE" == "" ]; then
     log "Must provide a Node Type value: master, client, or data" >&2
     exit 1
 fi
 
-log "Installing DataDog plugin onto this machine using API Key [$1]"
+#########################
+# Constants
+#########################
+
+ELASTIC_CONFIG_FILE=/etc/datadog-agent/conf.d/elastic.d/elastic.yaml
+DD_AGENT=/var/tmp/install_datadog_agent.sh
+
+#########################
+# Execution
+#########################
+
+log "Installing DataDog plugin onto this machine using API Key [$API_KEY]"
 
 wget -q https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh -O $DD_AGENT
 chmod +x $DD_AGENT
-DD_API_KEY=$1 $DD_AGENT
+DD_API_KEY=$API_KEY $DD_AGENT
 rm /etc/datadog-agent/conf.d/elastic.d/auto_conf.yaml
 
-echo "ad_identifiers:
```

Comment:
We don't need `ad_identifiers` anymore?

Reply:
that's for containers, and we don't use docker